vnext/Microsoft.ReactNative/Fabric/Composition/TooltipService.cpp

```diff
@@ -256,19 +255,21 @@ void TooltipTracker::ShowTooltip(const winrt::Microsoft::ReactNative::ComponentV
     facebook::react::TextLayoutManager::GetTextLayout(
         tooltipData->attributedString, {} /*paragraphAttributes*/, layoutConstraints, tooltipData->textLayout);
 
-    DWRITE_TEXT_METRICS tm;
-    winrt::check_hresult(tooltipData->textLayout->GetMetrics(&tm));
+    if (tooltipData->textLayout)
+    {
+      DWRITE_TEXT_METRICS tm;
+      winrt::check_hresult(tooltipData->textLayout->GetMetrics(&tm));
 
-    tooltipData->width =
-        static_cast<int>(tm.width + ((tooltipHorizontalPadding + tooltipHorizontalPadding) * scaleFactor));
-    tooltipData->height = static_cast<int>(tm.height + ((tooltipTopPadding + tooltipBottomPadding) * scaleFactor));
+      tooltipData->width =
+        static_cast<int>((tm.width + tooltipHorizontalPadding + tooltipHorizontalPadding) * scaleFactor);
```

Comment:
this seems a little perilous that the pattern `IDWriteTextLayout->GetMetrics` is implicitly or explicitly pinned to 100% display scale factor -- does it include text scaling but not display scaling?

Is the issue just getting the general size we need to avoid the clipping, so we're guessing a bit what the final size will be from DWrite?  Technically text scaling isn't guaranteed to be linear, but I believe that's mostly in the finer details of rasterizing, ligatures, subpixel rendering kinds of things. Any _normal_ font shouldn't cause distinctly different sizes at higher DPI, but should render better with the higher density.

Reply:
We adjust the text font size based on text scaling here https://github.com/microsoft/react-native-windows/blob/e20af4d3282101a02f7054f198e606a0596da356/vnext/Microsoft.ReactNative/Fabric/platform/react/renderer/textlayoutmanager/TextLayoutManager.cpp#L45 when creating the TextLayout.

I dont see any mechanism within TextLayout to account for the scale factor?  -- The scale factor does seem to be a factor until we render the text (at which point I think it just gets it from the render targets set DPI (which we do apply the scale factor to).